submissions/BlueLamantine/oop/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/BlueLamantine/a-tiny-JS-world
+   Web app: https://bluelamantine.github.io//a-tiny-JS-world/
+   */
+  class General {
+   constructor({ name, gender, friends }) {
+     this.name = name;
+     this.gender = gender;
+     this.friends = Array.isArray(friends) ? friends : 'no friends';
+   }
+ }
+ 
+ class Human extends General {
+   constructor({ name, gender, friends, voise }) {
+     super({ name, gender, friends });
+     this.voise = voise;
+     this.species = 'human';
+     this.legs = 2;
+     this.hands = 2;
+   }
+ 
+   getGender() {
+     return {
+       male: ['He', 'his'],
+       female: ['She', 'her'],
+     };
+   }
+ 
+   getInfo() {
+     return (
+       `This ${this.species} is a ${this.gender}, ` +
+       `${this.getGender()[this.gender][1]} name is ${this.name}. ` +
+       `${this.getGender()[this.gender][0]} has ${this.legs} legs and ${
+         this.hands
+       } hands. ` +
+       `${this.getGender()[this.gender][0]} says \'${
+         this.voise
+       }\'. Friend with : ${this.friends}`
+     );
+   }
+ }
+ 
+ class Man extends Human {
+   constructor({ name, friends }) {
+     super({ name, gender: 'male', friends, voise: 'What a wonderful world!' });
+   }
+ }
+ 
+ class Woman extends Human {
+   constructor({ name, friends }) {
+     super({
+       name,
+       gender: 'female',
+       friends,
+       voise: 'Feminism is in the air!',
+     });
+   }
+ }
+ 
+ class Pet extends General {
+   constructor({ name, gender, friends, species }) {
+     super({ name, gender, friends });
+     this.species = species;
+     this.paws = 4;
+   }
+ 
+   getInfo() {
+     return (
+       `This ${this.species} is a ${this.gender}, its name is ${this.name}. ` +
+       `The ${this.species} has ${this.paws} paws. Makes a sound \'${this.sound}\'. ` +
+       `Social with : ${this.friends}`
+     );
+   }
+ }
+ 
+ class Cat extends Pet {
+   constructor({ name, gender, friends }) {
+     super({ name, gender, friends, species: 'cat' });
+     this.sound = 'Meow Purrrr Purrr';
+   }
+ }
+ 
+ class Dog extends Pet {
+   constructor({ name, gender, friends }) {
+     super({ name, gender, friends, species: 'dog' });
+     this.sound = 'Woof woof';
```

Comment:
Both cats and dogs have a shared property `sound`. 
Whenever two sibling classes have a similar property it should be generalized to a base class. To the top base class if needed (like `name`).

The point of having similar things generalized is that any handling of those happens in a single place.

Please check your entire codebase to follow this principle.

Reply:
Now it is clear for me, removed that and some other duplications !